File: data-client/src/main/java/org/wikipedia/dataclient/mwapi/MwPostResponse.java
package org.wikipedia.dataclient.mwapi;

import androidx.annotation.Nullable;

public class MwPostResponse extends MwResponse {
    private int success;

    public boolean success(@Nullable String result) {
        return "success".equals(result);
    }

    public int getSuccessVal() {
        return success;
    }
}

File: app/src/main/java/fr/free/nrw/commons/wikidata/mwapi/MwPostResponse.kt
package fr.free.nrw.commons.wikidata.mwapi

import com.google.gson.annotations.SerializedName

open class MwPostResponse : MwResponse() {
    @SerializedName("success")
    val successVal: Int = 0

    fun success(result: String?): Boolean =
        "success" == result
}

